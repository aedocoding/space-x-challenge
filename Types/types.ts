export interface ILaunchListItem {
  name: string;
  details: string;
  image: string;
}
export type ScreenType = "LaunchList" | "CreateLaunch";
