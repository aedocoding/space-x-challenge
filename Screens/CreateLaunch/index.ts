export{ default as CreateLaunch} from './CreateLaunch'