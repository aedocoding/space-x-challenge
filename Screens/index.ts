export {LaunchDetails} from './LaunchDetails'
export {CreateLaunch} from './CreateLaunch'
export {LaunchList} from './LaunchList'