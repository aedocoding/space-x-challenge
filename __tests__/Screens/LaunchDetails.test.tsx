import renderer from "react-test-renderer";
import { ILaunchListItem } from "../../Types/types";
import { LaunchDetails } from "../../Screens";
jest.useFakeTimers();

describe("<LaunchDetails />", () => {
  const mockOpenLaunchState: boolean = false;
  const mockOpenLaunchSetter = (param: boolean) => {};
  const mockLaunchListItem: ILaunchListItem = {
    name: "Mock Launch",
    image:
      "http://www.nasa.gov/sites/default/files/thumbnails/image/42374725534_a653d1b7cb_k.jpg",
    details: "Mock Details",
  };
  it("has 1 child", () => {
    const tree = renderer
      .create(
        <LaunchDetails
          launchListItem={mockLaunchListItem}
          openLaunchState={mockOpenLaunchState}
          openLaunchSetter={mockOpenLaunchSetter}
        />
      )
      .toJSON();
    expect(tree.children.length).toBe(4);
  });
  test("renders correctly", () => {
    const tree = renderer
      .create(
        <LaunchDetails
          launchListItem={mockLaunchListItem}
          openLaunchState={mockOpenLaunchState}
          openLaunchSetter={mockOpenLaunchSetter}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
