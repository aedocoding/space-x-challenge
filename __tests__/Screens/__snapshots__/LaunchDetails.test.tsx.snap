// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LaunchDetails /> renders correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  visible={false}
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "http://www.nasa.gov/sites/default/files/thumbnails/image/42374725534_a653d1b7cb_k.jpg",
      }
    }
    style={
      Object {
        "height": 400,
        "width": "100%",
      }
    }
  />
  <Text>
    Mock Launch
  </Text>
  <Text>
    Mock Details
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Close
      </Text>
    </View>
  </View>
</Modal>
`;
